File: Data_Structure/Linear/Stack/Java/Stack.java
class Stack<T>{
    private node<T> top;
    private int count;
    
    public Stack() {
        top = null;
        count = 0;
    }
    
    public void push(T val) {
        node<T> newVal = new node<T>(val);
        newVal.next = top;
        top = newVal;
        count++;
    }

    public boolean isEmpty() {
        return top == null;
    }

    public void pop() {
        if (!isEmpty()) {
            node<T> last = top;
            top = top.next;
            last.next = null;  
            count--;
        }
    }

    public int size() {
        return count;
    }
    public T getTop() {
        return top.getValue();
    }
    public void print() {
        node<T> start = top;
        while (start!= null) {
            System.out.println(start.getValue());
            start = start.next;
        }
    }
}

File: Data_Structure/Linear/Stack/C++/stack.cpp
#include <bits/stdc++.h>
using namespace std;
template <class T>
class Stack
{
private:
    struct node
    {
        T val;
        node *next;
    };
    node *top;
    int count;

public:
    Stack()
    {
        top = NULL;
        count = 0;
    }
    void push(T val)
    {
        node *newVal = new node;
        newVal->val = val;
        newVal->next = top;
        top = newVal;
        count++;
    }
    void pop()
    {
        if (!isEmpty())
        {
            node *last = top;
            top = top->next;
            delete last;
            count--;
        }
    }
    bool isEmpty()
    {
        return top == NULL;
    }
    int size()
    {
        return count;
    }
    T getTop()
    {
        return top->val;
    }
    void print()
    {
        node *start = top;
        while (start != NULL)
        {
            cout << start->val << '\n';
            start = start->next;
        }
    }
    ~Stack() {
        while (!isEmpty())
        {
            pop();
        }
    };
};
int main()
{

    Stack<int> st;
    st.push(1);
    st.push(2);
    st.push(3);
    st.push(4);
    st.push(5);
    st.pop();
    st.print();
    cout <<"Size : " <<st.size() << '\n';
    cout<<"Is empty : " << st.isEmpty() << '\n';
    cout <<"Top : " <<st.getTop() << '\n';

    return 0;
}